packages/dom/src/document.rs

```diff
@@ -217,7 +222,9 @@ impl Document {
 
         let node = &self.nodes[node_id];
         let node_child_idx = node.child_idx;
-        let parent_id = node.parent.unwrap();
+
+        // Get this node's parent, or the root node if it has none.
+        let parent_id = node.parent.unwrap_or_default();
```

Comment:
This doesn't seem right. Only the root node should have no parent. And we wouldn't want to insert a child of the root node instead of a sibling. What is the use case here? Replacing the root node?

Reply:
@matthunz This is the only thing blocking this PR as far as I'm concerned. I'd like to understand why this is required. It seems wrong that `WriteMutations::insert_nodes_after` would be called on a node without a parent.  @ealmloff Any thoughts?

Reply:
I don't think this should be necessary. The root node should be `ElementId(0)` and every node except that node should have a parent in dioxus mutations. Dioxus will never insert a node after or before `ElementId(0)`